addons/l10n_de/models/res_company.py

```diff
@@ -17,6 +17,11 @@ class ResCompany(models.Model):
     )
     l10n_de_widnr = fields.Char(string="W-IdNr.", help="Business identification number.", tracking=True)
 
+    def write(self, vals):
+        if self.account_fiscal_country_id.code == 'DE' and 'account_fiscal_country_id' in vals and self.env['account.move'].search_count([('company_id', '=', self.id)], limit=1):
```

Comment:
* check the cheapest conditions before (no need to fetch the country if we are not writing on it)
* avoid lines too long (they still are a bit long in my suggestion...)
* `self` can have multiple records
```suggestion
        if (
            'account_fiscal_country_id' in vals 
            and (german_companies := self.filtered(lambda c: c.account_fiscal_country_id.code == 'DE'))
            and self.env['account.move'].search_count([('company_id', 'in', german_companies.ids)], limit=1)
        ):
```